sklearn/utils/estimator_checks.py

```diff
@@ -355,6 +357,26 @@ def _generate_class_checks(Estimator):
     yield from _generate_instance_checks(name, estimator)
 
 
+def _check_tags_to_skip(check):
+    """Wrap check and xfail or skip test based on estimator tag: _xfail_test"""
+
+    @wraps(check)
+    def wrapper(name, estimator_orig):
+        xfail_checks = _safe_tags(estimator_orig, '_xfail_test')
+        check_name = _set_check_estimator_ids(check).split("(", maxsplit=1)[0]
+        if xfail_checks and check_name in xfail_checks:
+            msg = xfail_checks[check_name]
+            try:
+                import pytest
+                pytest.xfail(msg)
```

Comment:
I agree we need something like it, and indeed raising SkipTest on xfail was missing in the original PR, I added it differently here https://github.com/scikit-learn/scikit-learn/pull/16507#discussion_r382332833.

However, with pytest installed wouldn't this stop exectution with an xfail exception, and so we will never get xpass. Paybe there is some way around it with `pytest.param(..., marks=pytest.mark.xfail())` not sure.

Reply:
Updated this PR with another approach. The `parametrize` is now generated in such a way that the xfails are marked.

Reply:
Thanks have you checked that this indeed runs the tests (I know it should) i.e. that marking a passing test with xfail results in an XPASS?